packages/i18next/src/lib/I18nextHandler.ts

```diff
@@ -0,0 +1,146 @@
+import { getRootData } from '@sapphire/pieces';
+import { Awaited, mergeDefault } from '@sapphire/utilities';
+import { readdir, stat as pStat } from 'fs/promises';
+import i18next, { InitOptions, StringMap, TFunction, TOptions } from 'i18next';
+import Backend, { i18nextFsBackend } from 'i18next-fs-backend';
+import { join } from 'path';
+import type { I18nOptions } from './types/options';
+
+/**
+ * A generalised class for handling `i18next` JSON files and their discovery.
+ * @since 1.0.0
+ */
+export class I18nextHandler {
+	/**
+	 * Describes whether `I18nextHandler#init` has been run and languages are loaded in `I18nextHandler.languages`.
+	 * @since 1.0.0
+	 */
+	public languagesLoaded = false;
+
+	/**
+	 * A `Map` of `i18next` language functions keyed by their language code.
+	 * @since 1.0.0
+	 */
+	public readonly languages = new Map<string, TFunction>();
+
+	private readonly options?: I18nOptions;
+
+	private readonly languagesDir: string;
+	private readonly backendOptions: i18nextFsBackend.i18nextFsBackendOptions;
+
+	/**
+	 * @param options The options that `i18next`, `i18next-fs-backend`, and {@link I18nextHandler} should use.
+	 * @since 1.0.0
+	 * @constructor
+	 */
+	public constructor(options?: I18nOptions) {
+		this.options = options;
+		this.languagesDir = this.options?.defaultLanguageDirectory ?? join(getRootData().root, 'languages');
+
+		this.backendOptions = mergeDefault(
+			{
+				loadPath: join(this.languagesDir, '{{lng}}', '{{ns}}.json'),
+				addPath: this.languagesDir
+			} as i18nextFsBackend.i18nextFsBackendOptions,
+			this.options?.backend
+		);
+	}
+
+	/**
+	 * Intitialises the handler by loading in the namespaces, passing the data to i18next, and filling in the {@link I18nextHandler#languages}.
+	 * @since 1.0.0
+	 */
+	public async init() {
+		const { namespaces, languages } = await this.walkLanguageDirectory(this.languagesDir);
+
+		i18next.use(Backend);
+		await i18next.init(
+			mergeDefault(
```

Comment:
See earlier comment by Kyra: https://github.com/sapphire-project/plugins/pull/29#discussion_r549511026

Reply:
No mergeDefault will create issues, as we concluded on Discord. Until further notice this is resolved.